### PR TITLE
Handle more bytecodes.

### PR DESCRIPTION
**Pull request by ptersilie**
This gets us a bit further in the small sample trace we are using.

### CODE REVIEW

File: ykrt/src/compile/jitc_yk/trace_builder.rs [hunk 1]
```diff
@@ -328,7 +328,9 @@ impl<'a> TraceBuilder<'a> {
                 }
                 None => {
                     // UnmappableBlock block
-                    todo!();
+                    // Ignore for now. May be later used to make sense of the control flow. Though
```
Comment (vext01) [inline ykrt/src/compile/jitc_yk/trace_builder.rs:339]: "may be" -> "maybe"

File: ykrt/src/compile/jitc_yk/trace_builder.rs [hunk 2]
```diff
@@ -328,7 +328,9 @@ impl<'a> TraceBuilder<'a> {
                 }
                 None => {
                     // UnmappableBlock block
-                    todo!();
```
Comment (ltratt) [inline ykrt/src/compile/jitc_yk/trace_builder.rs:295]: Why have we removed this? I know we want to do one day, but we can't do so yet can we?

File: ykrt/src/compile/jitc_yk/aot_ir.rs [hunk 3]
```diff
@@ -678,6 +693,20 @@ pub(crate) struct FuncType {
     is_vararg: bool,
 }
 
+impl FuncType {
+    pub fn arg_ty_idxs(&self) -> &Vec<TypeIdx> {
```
Comment (ltratt) [inline ykrt/src/compile/jitc_yk/aot_ir.rs:697]: `pub(crate)` unless we *really* have to `pub` these (which I don't think we do).

File: ykrt/src/compile/jitc_yk/trace_builder.rs (hunk 1 above)
Comment (ltratt) [reply ykrt/src/compile/jitc_yk/trace_builder.rs:339]: I think it's correct as-is.

File: ykrt/src/compile/jitc_yk/trace_builder.rs (hunk 2 above)
Comment (vext01) [reply ykrt/src/compile/jitc_yk/trace_builder.rs:295]: No real test case will be testable if we crash out here.

**Comment by vext01** [comment]
LGTM, bar a typo.

File: ykrt/src/compile/jitc_yk/trace_builder.rs (hunk 2 above)
Comment (ltratt) [reply ykrt/src/compile/jitc_yk/trace_builder.rs:295]: Wouldn't it be better to leave it in until/unless that does happen?

Comment (ptersilie) [reply ykrt/src/compile/jitc_yk/trace_builder.rs:295]: If we don't ignore unmappable traces, we can't continue dealing with the current sample trace we got. In the future we would have to put outlining logic here probably.

File: ykrt/src/compile/jitc_yk/aot_ir.rs (hunk 3 above)
Comment (ptersilie) [reply ykrt/src/compile/jitc_yk/aot_ir.rs:697]: `pub(crate)` didn't work. Let me try again.

File: ykrt/src/compile/jitc_yk/trace_builder.rs (hunk 2 above)
Comment (ltratt) [reply ykrt/src/compile/jitc_yk/trace_builder.rs:295]: OK.

File: ykrt/src/compile/jitc_yk/aot_ir.rs (hunk 3 above)
Comment (ptersilie) [reply ykrt/src/compile/jitc_yk/aot_ir.rs]: I could have sworn that I was getting an error at some point, but maybe that was unrelated. Seems to be working now. Fixed in [c8ff9d9](https://github.com/ykjit/yk/pull/988/commits/c8ff9d93537605107934da4253109b6ce76beb20)

**Comment by ptersilie** [comment]
Ok to squash?

**Comment by ltratt** [comment]
Please squash.

**Comment by ptersilie** [comment]
Squashed.
